waiter/src/waiter/state.clj

```diff
@@ -325,24 +375,29 @@
 
 (defn handle-kill-instance-request
   "Handles a kill request."
-  [{:keys [id->instance instance-id->state] :as current-state} update-status-tag-fn
-   [{:keys [request-id] :as reason-map} resp-chan exclude-ids-set _]]
-  (let [{instance-id :id :as instance-to-offer}
-        (find-killable-instance id->instance instance-id->state #(not (contains? exclude-ids-set %)))]
-    (if instance-to-offer
-      {:current-state' (-> current-state 
-                           (assoc-in [:instance-id->request-id->use-reason-map instance-id request-id] reason-map)
-                           ; mark instance as locked if it is going to be killed.
-                           (update-in [:instance-id->state instance-id] update-status-tag-fn #(conj % :locked)))
-       :response-chan resp-chan
-       :response instance-to-offer}
+  [{:keys [id->instance instance-id->request-id->use-reason-map instance-id->state] :as current-state}
+   update-status-tag-fn lingering-request-threshold-ms [{:keys [request-id] :as reason-map} resp-chan exclude-ids-set _]]
+  (let [acceptable-instance-id? #(not (contains? exclude-ids-set %))
+        ;; look for idle killable instance first before force killing an expired instance
```

Comment:
Could `find-killable-instance` more naturally handle this preference, without an extra explicit call to `find-oldest-acceptable-expired-instance`?  It already maintains a priority over the order in which instances get killed.